src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int AMOUNT_INDEX = 1;
+    private static final int TYPE_INDEX = 0;
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String text = readFileAndCreateReport(fromFileName);
+        writeReportToFile(toFileName, text);
+    }
 
+    private String readFileAndCreateReport(String fromFile) {
```

Comment:
When you use `And` in the method name it means that you should divide logic into separate methods (one method - one responsibility). Let's first read the file and pass the data to another method.